roms_tools/setup/cdr_forcing.py

```diff
@@ -886,180 +720,254 @@ def from_yaml(cls, filepath: Union[str, Path]) -> "CDRVolumePointSource":
+    and ensures each release is present in the set of valid releases defined in `self.releases`. Invalid releases
+    are reported in the error message.
 
-            eta_max = self.grid.ds.sizes["eta_rho"] - 1
-            xi_max = self.grid.ds.sizes["xi_rho"] - 1
+    If `list_allowed` is set to `False`, only a single release name (string) will be accepted. Otherwise, a
+    list of release names is also acceptable.
+    """
 
-            if eta_rho in [0, eta_max] or xi_rho in [0, xi_max]:
-                raise ValueError(
-                    f"Release site '{release.name}' is located too close to the grid boundary. "
-                    "Place release location (lat, lon) away from grid boundaries."
-                )
+    # Ensure that a list of releases is only allowed if `list_allowed` is True
+    if not list_allowed and not isinstance(releases, str):
+        raise ValueError(
+            f"Only a single release name (string) is allowed. Got: {releases}"
+        )
 
-            if self.grid.ds.mask_rho[eta_rho, xi_rho].values == 0:
-                raise ValueError(
-                    f"Release site '{release.name}' is on land. "
-                    "Please provide coordinates (lat, lon) over ocean."
-                )
+    if isinstance(releases, str):
+        releases = [releases]  # Convert to list if a single string is provided
+    elif isinstance(releases, list):
+        if not all(isinstance(r, str) for r in releases):
+            raise ValueError("All elements in `releases` list must be strings.")
+    else:
+        raise ValueError(
+            "`releases` should be a string (single release name) or a list of strings (release names)."
+        )
 
-            if self.grid.ds.h[eta_rho, xi_rho].values < release.depth:
-                raise ValueError(
-                    f"Release site '{release.name}' lies below the seafloor. "
-                    f"Seafloor depth is {self.grid.ds.h[eta_rho, xi_rho].values:.2f} m, "
-                    f"but requested depth is {release.depth:.2f} m. Adjust depth or location (lat, lon)."
-                )
+    # Validate that the specified releases exist in self.releases
+    invalid_releases = [
+        release for release in releases if release not in valid_releases
+    ]
+    if invalid_releases:
+        raise ValueError(f"Invalid releases: {', '.join(invalid_releases)}")
 
-        else:
-            logging.warning(
-                "Grid not provided: cannot verify whether the specified lat/lon/depth location is within the domain or on land. "
-                "Please check manually or provide a grid when instantiating the class."
-            )
 
-    def _validate_release_input(self, releases, list_allowed=True):
-        """Validates the input for release names in plotting methods to ensure they are
-        in an acceptable format and exist within the set of valid releases.
+def _get_release_colors(valid_releases):
```

Comment:
Consider reorganizing plotting methods into a new module to avoid importing unused packages

Reply:
We should do the same thing for all other roms-tools classes. Maybe in a future PR?